src/instruments/src/EFB/Presets/Widgets/LightPresets.tsx

```diff
@@ -0,0 +1,177 @@
+import { useSimVar } from '@instruments/common/simVars';
+import { toast } from 'react-toastify';
+import { usePersistentProperty } from '@instruments/common/persistence';
+import { ScrollableContainer } from '../../UtilComponents/ScrollableContainer';
+import { SimpleInput } from '../../UtilComponents/Form/SimpleInput/SimpleInput';
+
+export const LightPresets = () => {
+    // Manage names for presets in EFB only and always map them to the preset IDs used in the
+    // WASM implementation.
+    const [storedNames, setStoredNames] = usePersistentProperty('LIGHT_PRESET_NAMES', '');
+    const [namesMap, setNamesMap] = useState(new Map());
+
+    // called by the SinglePreset Component to get its assigned name
+    const getPresetName = (presetID: number): string => namesMap.get(presetID);
+
+    // Called by the SinglePreset Component to store its preset name after a name change
+    const storePresetName = (presetID: number, name: string) => {
+        namesMap.set(presetID, name);
+        const tmpJson = JSON.stringify(namesMap, replacer);
+        setStoredNames(tmpJson);
+    };
+
+    // Used by JSON.stringify for converting a Map to a Json string
+    function replacer(key, value) {
+        if (value instanceof Map) {
+            return {
+                dataType: 'Map',
+                value: Array.from(value.entries()), // or with spread: value: [...value]
+            };
+        }
+        return value;
+    }
+
+    // Used by JSON.parse for converting a Json string to a Map
+    function reviver(key, value) {
+        if (typeof value === 'object' && value !== null) {
+            if (value.dataType === 'Map') {
+                return new Map(value.value);
+            }
+        }
+        return value;
+    }
+
+    // Called once to initially load the preset names map from the persistent store
+    useEffect(() => {
+        try {
+            const newValue = JSON.parse(storedNames, reviver);
+            setNamesMap(newValue);
+        } catch {
+            setNamesMap(new Map());
+        }
+    }, []);
+
+    return (
+        <div className="p-4 mt-4 mb-4 h-content-section-reduced rounded-lg border-2 border-theme-accent">
+            <ScrollableContainer height={52}>
+                <div className="grid grid-cols-1 grid-rows-5 grid-flow-row gap-4">
+                    {[1, 2, 3, 4, 5, 6].map((i) => (
+                        <SinglePreset
+                            presetID={i}
+                            getPresetName={getPresetName}
+                            storePresetName={storePresetName}
+                            namesMap={namesMap}
+                        />
+                    ))}
+                </div>
+            </ScrollableContainer>
+        </div>
+    );
+};
+
+// One line of preset with ID, name, load and save
+const SinglePreset = (props: { presetID: number, getPresetName: (arg0: number) => string, storePresetName: (arg0: number, arg1: string) => void, namesMap: any }) => {
```

Comment:
Make sure to get this into its own interface and destructure props!

Reply:
Can you explain why this would be better? I really like to learn the reasoning. 
Adding an interface which is only used one single time sounds counter intuitive.